Snail/Observable.swift

```diff
@@ -216,6 +216,24 @@ public class Observable<T>: ObservableType {
         return observable
     }
 
+    public func take(first count: UInt) -> Observable<T> {
+        let observable = Observable<T>()
+        var taken = 0
+
+        subscribe(onNext: {
+            if taken < count {
+                observable.on(.next($0))
+                taken += 1
+            }
```

Comment:
Should we close the observable after taken is == to count?

Reply:
I know the test below tests that `.done` gets sent, but I don't see how it works here.

Reply:
This is a good point! Think it makes sense to send a `.completed` event when the observable has taken all values